### DISCUSSION TITLE
Atomicity of patching identities

### DISCUSSION THREAD

**Discussion by ariep** [ory/kratos#3641] [Q&A]
Could it be that PATCHing kratos identities is not atomic? I have a situation where two calls are made at almost the same time, both requesting a change to the `metadata_admin` property, and sometimes only one of the two goes through. Looking at the kratos logs for one of these times, the second request comes in before the first one is completed, so if kratos does a naive read/modify/write without any locking, I can see how that happens.

I would consider this a bug, not sure if everyone agrees.

**Comment by vinckr** [comment]
Hello @ariep 

It should be for traits but probably not for metadata. I have not tested this however.
